extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java

```diff
@@ -281,6 +283,16 @@ private void checkProperConfigAccessInProvider(AnnotationInstance instance) {
         }
     }
 
+    private void checkProperConstructorInProvider(AnnotationInstance i) {
+        ClassInfo targetClass = i.target().asClass();
+        if (!targetClass.hasNoArgsConstructor()
+                || targetClass.methods().stream().filter(m -> m.name().equals("<init>")).count() > 1) {
+            LOGGER.warn(
+                    "Classes annotated with @Provider should have a single, no-argument constructor, otherwise dependency injection won't work properly. Offending class is "
```

Comment:
The injection works fine if I add `@ApplicationScoped` to the filter (as described in the original issue). Why not use following text: 

```
Classes annotated with @Provider without @ApplicationScoped annotation should have a single, no-argument constructor, otherwise dependency injection won't work properly.
```

Has the if-statement to cover `@ApplicationScoped` as well?

Reply:
Now I'm a bit confused. @geoand We should probably verify whether and where constructor injection works... it could be that CDI injection works but JAX-RS injection does not work. I don't remember ;-).

Reply:
I couldn't get it to work reliably even with `@Singleton`, so I think it's best to play it safe.